Sources/ComposableArchitecture/Documentation.docc/Tutorials/BuildingSyncUps/05-PersistingSyncUps/PersistingSyncUps-01-code-0004.swift

```diff
@@ -1,18 +1,62 @@
 import ComposableArchitecture
 import SwiftUI
 
-@main
-struct SyncUpsApp: App {
-  @MainActor
-  static let store = Store(initialState: SyncUpsList.State()) {
-    SyncUpsList()
+@Reducer
+struct SyncUpsList {
+  @ObservableState
+  struct State: Equatable {
+    @Presents var addSyncUp: SyncUpForm.State?
+    @Shared(.syncUps) var syncUps: IdentifiedArrayOf<SyncUp> = []
   }
+  enum Action {
+    case addSyncUpButtonTapped
+    case addSyncUp(PresentationAction<SyncUpForm.Action>)
+    case confirmAddButtonTapped
+    case discardButtonTapped
+    case onDelete(IndexSet)
+    case syncUpTapped(id: SyncUp.ID)
+  }
+  var body: some ReducerOf<Self> {
+    Reduce { state, action in
+      switch action {
+      case .addSyncUpButtonTapped:
+        state.addSyncUp = SyncUpForm.State(syncUp: SyncUp(id: SyncUp.ID()))
+        return .none
+
+      case .addSyncUp:
+        return .none
+
+      case .confirmAddButtonTapped:
+        guard let newSyncUp = state.addSyncUp?.syncUp
+        else { return .none }
+        state.addSyncUp = nil
+        state.syncUps.append(newSyncUp)
+        return .none
+
+      case .discardButtonTapped:
+        state.addSyncUp = nil
+        return .none
+
+      case let .onDelete(indexSet):
+        state.syncUps.remove(atOffsets: indexSet)
+        return .none
 
-  var body: some Scene {
-    WindowGroup {
-      NavigationStack {
-        SyncUpsListView(store: Self.store)
+      case .syncUpTapped:
+        return .none
       }
     }
+    .ifLet(\.$addSyncUp, action: \.addSyncUp) {
+      SyncUpForm()
+    }
+  }
+}
+
+extension PersistenceReaderKey
```

Comment:
Changes regarding these files add one more step and some tutorial commentary to go from 
```swift
extension URL {
  static let syncUps = Self.documentsDirectory.appending(component: "sync-ups.json")
}
```
to
```swift
extension PersistenceReaderKey
where Self == PersistenceKeyDefault<FileStorageKey<IdentifiedArrayOf<SyncUp>>> {
  static var syncUps: Self {
    PersistenceKeyDefault(
      .fileStorage(.documentsDirectory.appending(component: "sync-ups.json")),
      []
    )
  }
}
```
In the current version of the tutorial, later on, this shorthand version is used, so we might as well just standardize on this throughout the tutorial, and this ensures that later parts of the tutorial compile.